components/script/dom/bindings/function.rs

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![allow(unsafe_code)]
+
+use std::ffi::c_char;
+use std::rc::Rc;
+
+use js::jsapi::{JSContext, JSObject, JS_GetFunctionObject, JS_NewFunction, Value};
+
+use crate::dom::bindings::codegen::Bindings::FunctionBinding::Function;
+use crate::dom::types::GlobalScope;
+
+type NativeFunction = unsafe extern "C" fn(*mut JSContext, u32, *mut Value) -> bool;
+
+pub struct FunctionBinding {}
+
+impl FunctionBinding {
```

Comment:
@gterzian @wusyong FYI.

Reply:
I was just thinking extend the Fuction type in codegen bindings. But I'm okay with any as long as the method is provided.

Reply:
```suggestion
impl Function {
```

Let's just extend Function then. Creating another type could confuse people.

Reply:
I used macro and removed this struct